aiogram/aiogram
# Add new FSM strategy CHAT_TOPIC strategy.py

# Description

The new FSM strategy, which sets the state for the entire topic in the chat, also works in private messages and regular groups without topics.

aiogram did not have an FSM strategy that would set state only for a specific topic in the chat. This strategy is quite reasonable and it has many applications. 
The work of state in private messages remains unchanged
The work of state in groups / super groups without topics, state will be established for all groups
The work of state in groups /super groups with topics, State will be established only for one topic, other topics will not have a state until it is installed, state can be set separately for each topic, and state_data will be separate in all topics where state is installed.


## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

- Run tests/test_fsm/test_strategy.py
- You can run bot with FSMStrategy.CHAT_TOPIC, and make sure that the strategy is working correctly:

   - Check the work of state in private messages
   - Check the work of state in groups/super groups without topics, State should be set for the whole group, since the topic value will be None
   - Check the work of state in groups/super groups with topics, State should be installed only for one topic, other topics will not have a state until it is installed, state can be set separately for each topic, and state_data for should be separate in all topics where state is installed.


Test Configuration:
* Operating System: windows 11
* Python version: 3.11

# Checklist:
